packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/sqs.js

```diff
@@ -254,6 +257,13 @@ class InstanaAWSSQS extends InstanaAWSProduct {
               // _synchronously_ in the same event loop tick. See commit message for details.
               span.cancel();
             }
+
+            // NOTE: attach the async context to continue with it if sqs-consumer is used, see index.js
+            //       sqs-consumer messages are dependend on the customer's `handleMessage` function
+            data.Messages.forEach(message => {
```

Comment:
https://github.com/bbc/sqs-consumer/blob/main/src/consumer.ts#L193

`receiveMessage` calls `this.sqs.send` where we can attach our async context. This is the only option.

https://github.com/bbc/sqs-consumer/blob/main/src/consumer.ts#L366

`executeHandler` calls the customer's `handleMessage` fn. This is the place where we need to wait for the result.